### DISCUSSION TITLE
Integrating with a plugin

### DISCUSSION THREAD

**Discussion by KRTirtho** [fzyzcjy/flutter_rust_bridge#460] [Q&A]
I've been trying to use flutter_rust_bridge for a plugin but it seems this project only supports integration with flutter application. Everything works if used inside a Flutter Application directly instead of a plugin abstraction

The `flutter run` command fails inside `<plugin-folder>/example` with `Failed to load dynamic library 'lib<crate_name>.so'` Exception
There's `lib<plugin-name>_plugin.so` but no `lib<crate_name>.so` inside `example/build/linux/x64/debug/bundle/lib`

Please help me with these issue. How can I bundle/link/import/load Rust's compiled shared library used inside the plugin into the application bundle directory?

**Comment by fzyzcjy** [comment]
Personally I integrate my rust code with a flutter plugin, and use that plugin in my app. This bridge works perfectly fine for that.

Just the tutorial does not demonstrate it currently.

Could you please provide more details and make it an issue?

/cc @Desdaemon who wrote the wonderful tutorial part.

**Comment by KRTirtho** [comment]
@fzyzcjy Thanks for the Super Fast Response
Of course, I'll open an issue with all the details but can you give me a link to one of those apps of yours that uses a plugin that uses flutter_rust_bridge?

**Reply by fzyzcjy** [reply]
@KRTirtho It is not public yet, neither open-sourced. But it is about 200k lines of Flutter code, and flutter+rust runs smoothly for me. What do you want to know?

**Reply by KRTirtho** [reply]
Ok, no problem then
